R/test.r

```diff
@@ -37,7 +37,10 @@ test <- function(pkg = ".", filter = NULL, ...) {
   }
 
   # Need to attach testthat so that (e.g.) context() is available
-  library(testthat, quietly = TRUE)
+  # Update package dependency to avoid explicit require() call (#798)
+  pkg$depends <- paste0("testthat, ", pkg$depends)
```

Comment:
Could you please elaborate? `testthat` is normally listed in `Suggests:` and by default would not be loaded by `load_all`.